### PR TITLE
Non Null Edges

### PR DESCRIPTION
**Pull request by benwilson512**

### CODE REVIEW

File: test/lib/absinthe/relay/connection_test.exs [hunk 1]
```diff
@@ -573,6 +574,15 @@ defmodule Absinthe.Relay.ConnectionTest do
     end
   end
 
+  describe "when defined with non_null: true" do
+    test "it will define edges and edge as non_null" do
+      always_present_pets_edge =
+        CustomConnectionAndEdgeFieldsSchema.__absinthe_type__(:always_present_pets_edge)
+
+      assert
```
Comment (benwilson512) [inline test/lib/absinthe/relay/connection_test.exs:582]: @jeroenvisser101 I created a PR based on your comment, this here is causing it to fail.

Comment (jeroenvisser101) [reply test/lib/absinthe/relay/connection_test.exs:582]: Yeah, that's right, the tests were still pending, the plan was to do assertions on the type to check that it had `non_null` added to it. Any hints on how to do this?

There are also 3 possible ways of defining non_null (separately with non_null_edge, non_null_edges and non_null, all of which should be tested).

File: test/lib/absinthe/relay/connection_test.exs [hunk 2]
```diff
@@ -188,7 +189,7 @@ defmodule Absinthe.Relay.ConnectionTest do
       end
 
       @desc "The favorite pets for a person"
-      connection field :favorite_pets, connection: :favorite_pets do
+      connection field :favorite_pets, node_type: :pet, connection: :favorite_pets do
```
Comment (jeroenvisser101) [inline test/lib/absinthe/relay/connection_test.exs:192]: I'm also not 100% sure, but I think node_type always had to be defined, and I'm quite sure that the node type here was never valid but the `connection field` didn't check this(?) I remember this was causing issues while I was writing the test, but can't remember why.

Is this a valid syntax (without specifying `node_type`)?

**Comment by jeroenvisser101** [comment]
@benwilson512 I added some tests, they aren't very elegant, but I think they cover all functionality. I also rebased it against master. Let me know if there's anything else :)

**Comment by hadfieldn** [comment]
@jeroenvisser101 @benwilson512 Really happy to see movement on this issue. Thanks so much!

File: lib/absinthe/relay/connection/notation.ex [hunk 3]
```diff
@@ -106,38 +106,38 @@ defmodule Absinthe.Relay.Connection.Notation do
   See the `edge` macro below for more information.
   """
   defmacro connection({:field, _, [identifier, attrs]}, do: block) when is_list(attrs) do
-    # do_connection_field(__CALLER__, identifier, naming_from_attrs!(attrs), field_attrs, block)
+    # do_connection_field(__CALLER__, identifier, Naming.from_attrs!(attrs), field_attrs, block)
```
[the remark this answers is not in the document]
Comment (jeroenvisser101) [reply lib/absinthe/relay/connection/notation.ex:109]: For sure!

[the remark this answers is not in the document]
Comment (jeroenvisser101) [reply lib/absinthe/relay/connection/notation.ex]: Done and rebased!

**Comment by jeroenvisser101** [comment]
Thanks!

**Comment by jeroenvisser101** [comment]
@benwilson512 do you see a chance to release a new version that includes this (and other) changes? Thanks :)

**Pull request merged**